### ISSUE TITLE
Gradient tools enhancement ideas

### ISSUE THREAD

**Issue by dsizzle** [HaikuArchives/ArtPaint#142]
The gradient part of the Flood fill tool could use a number of enhancements:

Easy:
- Color presets: Foreground to Background, Foreground to Transparent, Background to Transparent, Black to White, Black to Transparent, White to Transparent
- Flip gradient direction

Medium:
- Explicit/live color selection rather than only drag-and-drop

Maybe harder?:
- Different styles: Linear, radial, etc.
- More than two colors

**Comment by humdingerb** [comment]
WRT the colour selection (and other items of the above list), WonderBrush could be a nice example. Maybe even to borrow code from...:

![WB_gradient](https://user-images.githubusercontent.com/6107158/165027385-ec18032c-8610-4d0f-9803-52acde09f36d.png)

Clicking on a triangle would open the Color window to set the colour. As long as the triangle is selected (the one underlined), the colour keeps being updated live.
Double-click into the gradient bar creates a new colour point there. You can move the triangles around to flip colours or 'compress' the gradient.
One thing WB's gradient tool doesn't seem to allow: removing a triangle. That could be done either by a context menu, or dragging  the triangle far enough up or down to have it snap off the gradient bar.

**Comment by dsizzle** [comment]
Yes...that's exactly the kind of thing I had in mind. I'll have to check out WB for sure.

**Comment by dsizzle** [comment]
- Gradient angle should use the SHIFT modifier key to constrain it to 22.5 degree increments

**Comment by dsizzle** [comment]
- Different gradient interpolations - linear vs. perceptual

**Comment by humdingerb** [comment]
There are also "Cone" and "Diamond" styles.

Maybe having small greyscale icons for those styles will be better than adding more radio buttons. Would also be more visual than imagine what 'Diamond' may look like.

**Comment by humdingerb** [comment]
Hmm, I think you're right.
Oh well, let's keep icons at the back of our minds if space becomes scarce.

Saving gradients, esp. when they allow more than two colours, would be nice, too.
